plugins/idea/pluginBase/src/main/java/com/imaginea/kodebeagle/base/util/SearchUtils.java

```diff
@@ -202,16 +218,77 @@ public final String getProjectName(final String fileName) {
+    public final void updateRepoStarsMap(final String kbResultJson) {
+
+        JsonArray hitsArray = getJsonHitsArray(getJsonElements(kbResultJson));
+        hitsArray.toString();
         for (JsonElement hit : hitsArray) {
-            JsonObject sourceObject = hit.getAsJsonObject().getAsJsonObject(SOURCE);
-            if (sourceObject != null) {
-                String repoName = getProjectName(sourceObject.get("file").getAsString());
-                String score = sourceObject.get("score").getAsString();
-                windowObjects.getRepoStarsMap().put(repoName, score);
+
+            JsonObject hitObject = hit.getAsJsonObject();
+            hitObject.toString();
+            String score = hitObject.getAsJsonPrimitive(SCORE).getAsString();
+            String fileName = hitObject.getAsJsonPrimitive(FILE_NAME).getAsString();
+            String repoName = fileName.substring(0, fileName.indexOf(GITHUB_BLOB) -1);
+            windowObjects.getRepoStarsMap().put(repoName, score);
+
+        }
+    }
+
+    public final String getQueryJson(final Map<String, Set<String>> importsInLines,
+                                     final boolean includeMethods, final int from, final int size) {
+
+        SearchQuery searchQuery = new SearchQuery();
+        searchQuery.setFrom(from);
+        searchQuery.setSize(size);
+        Set<SearchQuery.QueryParam> queryParams = new HashSet<>();
+        for (String type : importsInLines.keySet()) {
+
+            Set<String> methods = importsInLines.get(type);
+            if (methods.size() == 0 || !includeMethods) {
+
+                queryParams.add(new SearchQuery.QueryParam(type, SearchQuery.Type.type));
+
+            } else {
+                for (String method : methods) {
+
+                    queryParams.add(new SearchQuery.QueryParam(type + DOT + method, SearchQuery.Type.method));
+
+                }
+
             }
         }
+        searchQuery.setQueries(queryParams);
+        return searchQuery.build();
     }
+
+
+    public final List<Integer> getLineNumbers(final Collection<String> imports,
+                                              final String tokens) {
+
+        List<Integer> lineNumbers = new ArrayList<Integer>();
+        JsonReader reader = new JsonReader(new StringReader(tokens));
+        reader.setLenient(true);
+        JsonArray tokensArray = new JsonParser().parse(reader).getAsJsonArray();
+        for (JsonElement token : tokensArray) {
+            JsonObject jObject = token.getAsJsonObject();
+            String typeName = jObject.getAsJsonPrimitive(TYPE_EXACT_NAME).getAsString();
+            if (imports.contains(typeName)) {
+
+                JsonArray propsArray = jObject.getAsJsonArray(PROPS);
+                for (JsonElement propsInfo : propsArray) {
+
+                    JsonArray lineNumbersArray =  propsInfo.getAsJsonObject().getAsJsonArray(LINES);
+                    for (JsonElement lineNumber: lineNumbersArray) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)